packages/blade/src/components/Input/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,158 @@
+  const inputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
+  const [shouldShowClearButton, setShouldShowClearButton] = React.useState(false);
+
+  React.useEffect(() => {
+    if (showClearButton && (value?.length || defaultValue?.length)) {
+      setShouldShowClearButton(true);
+    }
+  }, [value, defaultValue, showClearButton]);
+
+  const renderInteractionElement = (): React.ReactNode => {
+    if (shouldShowClearButton) {
+      return (
+        <Box paddingTop="spacing.3" marginTop="spacing.1">
+          <IconButton
+            icon={CloseIcon}
+            accessibilityLabel="Clear textarea content"
+            onClick={() => {
+              if (isEmpty(value) && inputRef.current) {
+                // when the input field is uncontrolled take the ref and clear the input and then call the onClearButtonClick function
+                if (isReactNative(inputRef.current)) {
+                  inputRef.current.clear();
+                  inputRef.current.focus();
+                } else if (inputRef.current instanceof HTMLTextAreaElement) {
+                  inputRef.current.value = '';
+                  inputRef.current.focus();
+                }
+              }
+              // if the input field is controlled just call the click handler and the value change shall be left upto the consumer
+              onClearButtonClick?.();
+              inputRef?.current?.focus();
+              setShouldShowClearButton(false);
+            }}
+          />
+        </Box>
+      );
+    }
+
+    return null;
+  };
+
+  return (
+    <BaseInput
+      as="textarea"
+      autoFocus={autoFocus}
+      ref={inputRef as React.Ref<HTMLInputElement>}
+      label={label}
+      labelPosition={labelPosition}
+      necessityIndicator={necessityIndicator}
+      errorText={errorText}
+      helpText={helpText}
+      successText={successText}
+      validationState={validationState}
+      isDisabled={isDisabled}
+      isRequired={isRequired}
+      name={name}
+      maxCharacters={maxCharacters}
+      onBlur={onBlur}
+      placeholder={placeholder}
+      interactionElement={renderInteractionElement()}
+      defaultValue={defaultValue}
+      value={value}
+      numberOfLines={numberOfLines}
+      onChange={({ name, value }) => {
+        if (showClearButton && value?.length) {
+          // show the clear button when the user starts typing in
+          setShouldShowClearButton(true);
+        }
+
+        if (shouldShowClearButton && !value?.length) {
+          // hide the clear button when the input field is empty
+          setShouldShowClearButton(false);
+        }
+
+        onChange?.({ name, value });
+      }}
+      trailingFooterSlot={(value) => {
+        return maxCharacters ? (
+          <Box marginTop="spacing.2" marginRight="spacing.1">
+            <CharacterCounter currentCount={value?.length ?? 0} maxCount={maxCharacters} />
+          </Box>
+        ) : null;
+      }}
+      id=""
```

Comment:
@kamleshchandnani why is this ID mandetory?

Reply:
to generate all the a11y ids specific to that field else everything will be default to default value of `base-input-***` since it's internal component let's be more specific and restrictive about what kind of ids will be generated else it'll be confusing to debug for consumers